connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -208,6 +228,22 @@ protected void initLoaders() {
         // Finally add parent/system loader.
         initPluginLoader(CLASSPATH_NAME);
         addAllAliases();
+        reportPluginConflicts();
+    }
+
+    //visible for testing
+    Set<String> reportPluginConflicts() {
+        Set<String> conflictPluginClasses = new HashSet<>();
+        for (Map.Entry<String, List<PluginDesc<?>>> entry : allAddedPlugins.entrySet()) {
+            String pluginClassName = entry.getKey();
+            List<PluginDesc<?>> pluginDescriptors = entry.getValue();
+            if (pluginDescriptors.size() > 1) {
+                PluginDesc<?> pluginDescInUse = pluginDescInUse(pluginClassName);
+                log.error("For plugin '{}', detected multiple copies '{}', this copy '{}' will be used.", pluginClassName, pluginDescriptors, pluginDescInUse);
+                conflictPluginClasses.add(pluginClassName);
+            }
+        }
+        return conflictPluginClasses;
```

Comment:
This could be rewritten a bit more compactly and a bit more functionally:
```suggestion
        return allAddedPlugins.entrySet().stream().filter(e -> e.getValue().size() > 1).map(e -> {
            String pluginClassName = e.getKey();
            PluginDesc<?> pluginDescInUse = pluginDescInUse(pluginClassName);
            List<PluginDesc<?>> ignoredPlugins = new ArrayList<>(e.getValue());
            ignoredPlugins.remove(pluginDescInUse);
            log.error("Detected multiple plugins contain '{}'; using {} and ignoring {}", pluginClassName, pluginDescInUse, ignoredPlugins);
            return pluginClassName;
        }).collect(Collectors.toSet());
```
My suggestion also includes a reworded error message to put the more meaningful information near the front, and computes the unused plugins and includes them at the end of the message, just in case there are multiple.

Reply:
Applied all the changes. @rhauch please check now.